### PR TITLE
Upsert (update) validation failing if document is being selected by id

### PR DESCRIPTION
**Pull request by dmitrijs-balcers**
Yesterday we encountered with an issue, which was preventing us to perform an `upsert` using such selector `{_id: id}`. The problem is that `collection2` is merging selector and modifier.

I wrote a test which is simulating the issue and also I attempted to perform a fix, by excluding `_id` from modifier. Tests are passing, but it would be nice if someone could check if this is not going to lead to some additional bugs.

Long story short:
This works: 
```
books.upsert({_id: id}, {$set: {title: "Ulysses", author: "James Joyce", copies: 1}}, {validate: false});
```
This didn't work: 
```
books.upsert({_id: id}, {$set: {title: "Ulysses", author: "James Joyce", copies: 1}});
```

### CODE REVIEW

**Comment by dmitrijs-balcers** [comment]
@aldeed is there any issue with this PR?

**Comment by aldeed** [comment]
Getting caught up. This looks good, thanks!
